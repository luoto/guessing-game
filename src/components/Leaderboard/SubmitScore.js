import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import stopPropagation from '../../helpers/stopPropagation';

const SubmitScoreWrapper = styled.div`
  visibility: ${props => (props.shouldShow ? 'visible' : 'hidden')};
`;

class SubmitScore extends Component {
  state = {
    username: ''
  };

  onChange = e => {
    const { name, value } = e.target;

    this.setState({
      [name]: value
    });
  };

  saveScore = e => {
    this.props.saveScore(this.state.username);
  };

  render() {
    return (
      <SubmitScoreWrapper shouldShow={this.props.shouldShow}>
        <input
          type="text"
          placeholder="username"
          name="username"
          value={this.state.value}
          onChange={this.onChange}
          onKeyDown={stopPropagation}
        />
        <button onClick={this.saveScore}>Submit</button>
      </SubmitScoreWrapper>
    );
  }
}

SubmitScore.proptypes = {
  saveScore: PropTypes.func
};

export default SubmitScore;
