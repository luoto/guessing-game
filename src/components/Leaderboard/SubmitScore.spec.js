import React from 'react';
import { mount } from 'enzyme';
import SubmitScore from './SubmitScore';

it('renders without crashing', () => {
  mount(<SubmitScore />);
});

it('shows', () => {
  let wrapper = mount(<SubmitScore shouldShow={true} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('visibility', 'visible');
});

it('does not show', () => {
  let wrapper = mount(<SubmitScore shouldShow={false} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('visibility', 'hidden');
});

it('should submit luoto', () => {
  const saveScore = jest.fn();
  let wrapper = mount(<SubmitScore shouldShow={true} saveScore={saveScore} />);
  wrapper
    .find('input')
    .simulate('change', { target: { name: 'username', value: 'luoto' } });
  wrapper.find('button').simulate('click');

  expect(saveScore).toBeCalledWith('luoto');
});
