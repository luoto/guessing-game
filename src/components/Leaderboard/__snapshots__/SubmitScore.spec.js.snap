// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not show 1`] = `
.c0 {
  visibility: hidden;
}

<SubmitScore
  shouldShow={false}
>
  <styled.div
    shouldShow={false}
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  visibility: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      shouldShow={false}
    >
      <div
        className="c0"
      >
        <input
          name="username"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="username"
          type="text"
        />
        <button
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </StyledComponent>
  </styled.div>
</SubmitScore>
`;

exports[`shows 1`] = `
.c0 {
  visibility: visible;
}

<SubmitScore
  shouldShow={true}
>
  <styled.div
    shouldShow={true}
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  visibility: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      shouldShow={true}
    >
      <div
        className="c0"
      >
        <input
          name="username"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="username"
          type="text"
        />
        <button
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </StyledComponent>
  </styled.div>
</SubmitScore>
`;
